packages/@react-spectrum/dialog/src/AlertDialog.tsx

```diff
@@ -65,48 +66,50 @@ function AlertDialog(props: SpectrumAlertDialogProps, ref: DOMRef) {
   }
 
   return (
-    <Dialog
-      UNSAFE_style={styleProps.style}
-      UNSAFE_className={classNames(styles, {[`spectrum-Dialog--${variant}`]: variant}, styleProps.className)}
-      isHidden={styleProps.hidden}
-      size="M"
-      role="alertdialog"
-      ref={ref}>
-      <Heading>{title}</Heading>
-      {(variant === 'error' || variant === 'warning') &&
-        <AlertMedium
-          slot="typeIcon"
-          aria-label={formatMessage('alert')} />
-      }
-      <Divider />
-      <Content>{children}</Content>
-      <ButtonGroup align="end">
-        {cancelLabel &&
-          <Button
-            variant="secondary"
-            onPress={() => chain(onClose(), onCancel())}
-            autoFocus={autoFocusButton === 'cancel'}>
-            {cancelLabel}
-          </Button>
+    <div role="none" onKeyDown={isKeyboardCancelDisabled ? (e) => e.stopPropagation() : () => {}}>
```

Comment:
Rather than stopping propagation here, I think we should add the `isKeyboardCancelDisabled` option to the `useOverlay` hook and check it in the if statement there: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/overlays/src/useOverlay.ts#L69-L75

Reply:
Makes sense! I wasn't sure how to add it to `useOverlay` without going through `DialogTrigger`. 

I've made the changes, moved it to be an initial prop of `DialogTrigger`, and included it for the other types of overlays as well.